api/lib/infrastructure/repositories/organization-repository.js

```diff
@@ -101,6 +101,13 @@ module.exports = {
       .then((organizations) => organizations.models.map(_toDomain));
   },
 
+  findByExternalId(UAI) {
```

Comment:
Je ne suis pas fan de l'attribut tout en majuscule. De plus, ce que l'on cherche dan l'organisation est bien un `externalId` même si dans le cadre d'une organisation SCO on peut parler d'un UAI (mais c'est un cas particulier). De plus, tu pourrais directement ici récupérer une organisation SCO (en ajoutant une claue WHERE sur le type et renommer la méthode en conséquence) et, grâce à un `require: true`, forcer la levée d'un exception si cette organisation n'est pas trouvée. De toute façon, que l'`externalId` ne soit pas trouvé ou que l'organisation récupérée ne soit pas SCO, l'API doit renvoyer un `NotFound`.

Reply:
vu en mob: on change to findScoOrganizationByUai(uai) en mettant une clause where, car le message est le même pour les deux cas ( notFound et organization not de type de SCO ).
Pour le find, il est utilisé car on peut avoir plusieurs organizations pour le même UAI ( y a des cas ou l'uai n'est pas unique). Autre raison, le find ne lève pas d'exceptions quand il trouve pas, il renvoie une liste vide ou une liste aves les occurrences retrouvées contrairement au Get qui est utilisé lors de filtre avec ids uniques.